### PR TITLE
Beekeeping 3.6

### LINKED ISSUE #18587: The Bee 'Exploit'
#### Description of issue
You can exploit bee reproduction to such a degree that you can crash some people's byonds, or at the very least make it a lag spiked mess. If you have one apiary, completely full of honey, completely full of forty bees, with nearby plants to support the apiary, if you lay down an empty apiary, they will colonize it and very soon fill it with 40 bees. This is a slow process, however.

The issue is that once you have multiple bee hives, it's no longer really a matter of one full bee hive colonizing a new bee hive. If you have 2 apiaries, completely full, and lay down 4 new empty apiaries, all of those colonies will now move to colonize these new apiaries: They pick the first one laid down, then move on as it's 'noted' to be 'colonized'. These 2 apiaries target the first new apiary, then the second, and so on. Because so many apiaries are colonizing at once, the old apiaries lose some bees(maybe ten? I don't know), however the new apiaries are instantly full and ready to colonize new apiaries. Assuming you have your source of plant nectar like stated above, you now have 4 completely full apiaries. If you lay down 4 empty apiaries, you will now very soon have 8 full apiaries. And when you lay down 8 empty, you will have 16, etc etc.

This is not so bad, but please note:
#### Difference between expected and actual behavior
When you ax an apiary, it angers some bees, but eventually expels all bees as the homeless type. They will wander around and do mostly nothing unless you net them up yourself and put them all back in the apiary.

What does this mean?

You have ten completely full apiaries. You drop ten empties. They fill up, and are soon completely full. Ax the newest ten apiaries. Don't put the homeless bees back. Put down the empty apiaries. The full apiaries will again colonize them, with the 400 homeless bees stuck wandering(these bees do not die of natural causes, they can't starve). Ax the newly colonized apiaries once they're full again, you now have 800 bees. Then 1,200, and so on.

#### Steps to reproduce 
1. Raise bees
2. Get the apiaries full.
3. Lay down new apiaries.
4. Wait for them to fill.
5. Ax them.
6. Don't bother cleaning up the homeless bees.
7. Ad nauseam or until everything crashes.

#### Specific information for locating
Bees

#### Length of time in which bug has been known to occur
Probably since the beeginning of time.

#### Server revision
Bleedin'

#### Issue bingo

Please check whatever applies. More checkboxes checked increase your chances of not being yelled at by every contributor.

- [x] Issue could be reproduced at least once
- [x] Issue could be reproduced by different players
- [x] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/d3athrow/vgstation13/issues)

----

Homeless bees need to seek out a new apiary after some time maybe, but more importantly, they need to die after a given amount of time without a home or something. Also it should be noted this wasn't causing the controllers to crash afaik- that would happen when I harvested the apiaries, laying down the apiaries to repeat this  wouldn't have done much since the controllers would crash at that point.

### PR DESCRIPTION
**Pull request by DeityLink**
Fixes #18587

image

image

I've done some testing to squash the most obvious bugs, but as always I'm expecting you guys to break my stuff. Have some fun.

:cl:
* experiment: BEEKEEPING 3.6 - I'm back again baby
* tweak: You no longer have to land the extinguisher spray right onto a bee swarm to calm them down. Calming bees has never been easier!
* tweak: Bees no longer remain angry forever! If they can't find a target for an extended period of time, they'll calm down. This doesn't apply to angry bee infestations.
* tweak: When an empty apiary is available, no more than one queen will swarm at once to populate it.
* rscadd: Homeless bees will die after about 10 minutes if left alone. Except queen bees, who'll try to recruit homeless workers to build colony. Their swarms will target empty apiaries in priority, but if they can't find any, they'll build their own from scratch! Each species has their own wild hive.
* rscadd: Wild hives work like normal ones, except that you cannot add queen bee packets or beezeez, or empty bees into them.
* rscadd: Added Hornets. They have a chance to spawn from infestation events, and traitor botanists may purchase Hornet Queen Packets for 2 TC each. Hornets have a chance to deal partial damage through protection, so handle carefully!
* rscadd: Beekeepers now start the game with a single bee packet in their pocket, so they don't have to wait for cargo to get to work.
* tweak: lots of back-end optimization.

### CODE REVIEW

**Comment by Exxion** [comment]
It looks like `empty_beehive()` still doesn't add the removed bees to `bees_outside_hive`

**Comment by DeityLink** [comment]
empty_beehive() is only used in instances where the hive is getting destroyed, so doing so would be kinda obsolete.

**Comment by Exxion** [comment]
No, it would not. That is the list the beehive uses to null the references to itself on the bees. Not adding them means the beehive fails to get GCed and causes the bug I mentioned in #18587.
It would also work if you nulled their `home` directly in `empty_beehive()`, but you have to do one or the other.

**Comment by DeityLink** [comment]
Well I double checked but the bees created by empty_beehive() are homeless, their home isn't set (which is the arg following loc on New() ).

    var/mob/living/simple_animal/bee/lastBees = getFromPool(/mob/living/simple_animal/bee,get_turf(src))

After some tests, the bees spawned upon beehive deconstruction/destruction were always homeless, and so where those who had left the hive previously. Please try reproducing the bug using this version. I may have fixed it by tweaking something else.

**Comment by Exxion** [comment]
The mob's home isn't set, but the individual datums still are, from what I can see. But I'll check this version once I'm done with what I'm working on

EDIT: Oh wait I'm a dumbass
OK I'm not sure why that bug happens but since I already have this checked out, I'll do some testing

EDIT: No I'm not
I was partially right
https://github.com/vgstation-coders/vgstation13/pull/18592/files#diff-ed51e636a1dd18ba75305038dc2395cdL331
It's setting the homes of the bee *datums* right there
I remembered incorrectly; it's not that it needs to *unset* the homes in that proc, it's that it needs to not *set* them

Also, I was mistaken about it happening when you hatchet the hive. It does not, since the bees are all emptied from the hive *before* `empty_beehive()` is called. The bug is specific to `empty_beehive()`.
I'm not sure how I observed it before, since when I did, I hatcheted every hive.

PROBABLY THE FINAL EDIT: I guess when I saw it happen before, some bees probably entered the hive during the `do_after()`, but it can also happen if the beehive's species is not `angery`
In this case, the bees are *called* homeless, but you can clearly see in VV that they still hold a reference to the destroyed hive.
Something in `Life()` seems to usually null it later, but it shouldn't *have* to.

**Comment by DeityLink** [comment]
Guess I was really tired, thank you for catching that. Bee datums created by empty_beehive() will no longer have a home.

File: code/datums/uplink_item.dm [hunk 1]
```diff
@@ -273,6 +273,13 @@ var/list/uplink_items = list()
 	cost = 4
 	job = list("Botanist")
 
+/datum/uplink_item/jobspecific/hornetqueen
+	name = "Hornet Queen Packet"
+	desc = "Place her into an apiary tray, add a few packs of BeezEez, then lay it inside your nemesis' office. Surprise guarranted. Protective gear won't be enough to shield you reliably from those."
```
Comment (MadmanMartian) [inline code/datums/uplink_item.dm]: guaranteed, not guarranted.

These, not those.

**Review by MadmanMartian — CHANGES_REQUESTED** [review]
Just a few grammar problems

File: code/datums/uplink_item.dm (hunk 1 above)
Comment (DeityLink) [reply code/datums/uplink_item.dm]: fix'd

**Review by MadmanMartian — APPROVED** [review]

**Pull request merged**